### ISSUE TITLE
hook regular expression matches wrong code

### ISSUE THREAD

**Issue by zokrezyl** [glumpy/glumpy#251]
the following code is not a hook, though the regular expression in the glumpy/gloo/parser.py (get_hooks) is wrongly matching it:

```
if (dist == 0. || dist > radii.y || (mag<1.01 && mag>0.99))
```
the re in question:
```    re_hooks = re.compile("""\<(?P<hook>\w+)
                              (\.(?P<subhook>.+))?
                              (\([^<>]+\))?\>""", re.VERBOSE )
```

**Comment by rougier** [comment]
Thanks for the report. What is the matching hook if the source you've given ?

**Comment by zokrezyl** [comment]
@rougier, I was not matching a hook, the parser matched a piece of shader code (see above) that it should't have. The parser is in "glumpy/gloo/parser.py" and the code that was wrongly matched is 

```
if (dist == 0. || dist > radii.y || (mag<1.01 && mag>0.99))
```
in the regular expression I believe one should be more restrictive, match only the allowed characters instead of anything but not <>

**Comment by rougier** [comment]
Oh yes, now I see. You're right. Maybe excluding spaces would be enough.

**Comment by zokrezyl** [comment]
Well, excluding spaces you would cover just part of the problem... I can remove the spaces from the above expression and you would be in the same trouble... I am not sure what is the grammar you are trying to match, but so far I sow you need alphanum+dot, right? Why not matching those?

**Comment by rougier** [comment]
I agree and problem is that I don't remember what I'm using such a pattern for matching hook name. I would need to do as you suggest and check if it break things or not. By looking at the shader library, it should be relatively easy to see what will break.
